src/language/ast.ts

```diff
@@ -360,6 +374,30 @@ export interface FieldNode {
   readonly arguments?: ReadonlyArray<ArgumentNode>;
   readonly directives?: ReadonlyArray<DirectiveNode>;
   readonly selectionSet?: SelectionSetNode;
+  readonly required?: SupportArrayNode | NullabilityModifierNode;
+}
+
+export interface RequiredModifierNode {
+  readonly kind: Kind.REQUIRED_DESIGNATOR;
+  readonly loc?: Location;
+  readonly element?: SupportArrayNode;
+}
+
+export interface OptionalModifierNode {
+  readonly kind: Kind.OPTIONAL_DESIGNATOR;
+  readonly loc?: Location;
+  readonly element?: SupportArrayNode;
+}
+
+// modifiers can be !, ? or []
+export type NullabilityModifierNode =
+  | RequiredModifierNode
+  | OptionalModifierNode;
+
+export interface SupportArrayNode {
```

Comment:
Why is this called SupportArrayNode instead of ListNullabilityNode or at least SupportListNode? In GraphQL we are always talking about lists, not arrays.

Reply:
The name was based on this comment: https://github.com/graphql/graphql-js/pull/3281#discussion_r754859308

I'm not really a fan of the name I picked either, and I'm down to change it before everything gets merged. `ListNullabilityNode` sounds good to me, so I'll go with that.